A bunch of fixes

dkorunic opened the pull request:
- fix Vagrant file private IPs (using publicly routable IP for private
  network is really a bad idea)
- use native shared folder filesystem (vboxsf and fallback on nfs only if
  really needed)
- tune MySQL for eZ by adding ez.cnf
- tune PHP FPM
- stricten PHP memory limits back to reasonably low defaults
- relax PHP timing limits
- install, configure and enable OpCache

dkorunic committed:
- fix Vagrant file private IPs (using publicly routable IP for private
  network is really a bad idea)
- use native shared folder filesystem (vboxsf and fallback on nfs only if
  really needed)
- tune MySQL for eZ by adding ez.cnf
- tune PHP FPM
- stricten PHP memory limits back to reasonably low defaults
- relax PHP timing limits
- install, configure and enable OpCache

emodric commented on provisioning/roles/php/tasks/main.yml:
```diff
@@ -50,7 +50,7 @@
 
 - name: Customized memory limits
-  copy: dest={{ php_mods_available }}/memory_limits.ini content="; configuration for memory limits\nmemory_limit=512M\nupload_max_filesize=1024M\npost_max_size=1024M\n"
+  copy: dest={{ php_mods_available }}/memory_limits.ini content="; configuration for memory limits\nmemory_limit = 160M\nupload_max_filesize = 1024M\npost_max_size = 1024M\n"
```
eZ Publish requires at least 256M minimum to work properly, and ideally 512M to work without hiccups.

dkorunic replied:
Definitely not recommended in any kind of real-life production with preforked PHP. Experience shows that 128M is okay, even 160M is rarely needed. 128 conns x 256M is really quite a lot even on mid-range servers, let alone 512M. Avoid doing that for non-CLI PHP, and even for CLI I haven't had a case where it's absolutely required.

emodric replied:
Consider that we're talking about development mode, where all caches are turned off, symfony debug toolbar is activated (which alone consumes A LOT of memory), and all debug traces and points are activated.

I will test with your recommended settings and let you know, but up until now, in development mode, we had poor results with anything below 256M.

dkorunic committed:
- fix /var/www/project mount options
- disable opcache for CLI PHP
- up the revalidate freq for development-only box
- PHP FPM from repo is not systemd-patched
- up the memory limits as Netgen requested
- add kernel role with procps/sysctl tuning

dkorunic committed:
- revert from Vboxsf/native shared folders to NFS due to performance
- enable all required/useful Apache2 modules
- tune Apache security and performance-wise

dkorunic commented:
OK I believe we've convered it all except disabling XDebug.

dkorunic committed:
- proper site.j2 for Apache vhosts to allow access, disregarding what
  Apache system configuration has

dkorunic committed:
- more NFS options, compatible with OSX and Linux

vranac commented on provisioning/roles/mysql/files/ez.cnf:
```diff
@@ -0,0 +1,15 @@
+# development-only setup, still light on memory but usable enough
```
Rename this file to ez-dev.cnf, because later you will probably add ez-prod.cnf or something along the lines.

vranac commented on provisioning/roles/mysql/tasks/main.yml:
```diff
@@ -34,6 +34,10 @@
+- name: Tune for eZ development
+  copy: src=ez.cnf dest=/etc/mysql/conf.d/ez.cnf owner=root group=root mode=0644
+  notify: restart mysql
+
```
Move this segment after the next one.
The flow should go
"Ensure mysql service is up"
"Tune for ez development"

Please add a switch
when: use_mysql_dev == true

Set the default value in https://github.com/dkorunic/vagrant-ansible-ezpublish/blob/master/provisioning/roles/mysql/defaults/main.yml 

Change https://github.com/dkorunic/vagrant-ansible-ezpublish/blob/master/provisioning/vagrant.yml#L11
to - { role: mysql, use_mysql_dev: true}

Add the description to https://github.com/dkorunic/vagrant-ansible-ezpublish/blob/master/provisioning/roles/mysql/README.md under variables

dkorunic committed:
- add required changes (ez.cnf -> ez-dev.cnf), add use_mysql_dev handling

dkorunic committed:
- add ez-prod.cnf for MySQL production values (still not templated)

dkorunic committed:
- enable multiverse repos (and multiverse security repos)
- refactor Apache install to itemized install
- disable dir indexes by default in ez vhost
- use fastcgi + FPM when php5-fpm is enabled
- refactor fpm and non-fpm installs to not install/overwrite each other
  with conflicting modules
- properly enable FPM support in Apache and use mod fcgi from multiverse

vranac commented:
:+1:

dkorunic committed:
Merge remote-tracking branch 'upstream/master'
